### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by Stephen-Arsenault**

### CODE REVIEW

File: lib/AccessControlClient.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  constructor(accessControlAppKey, accessControlAppSecret) {
+    this.appKey = accessControlAppKey;
+    this.appSecret = accessControlAppSecret; 
+    this.axios = axios.create({
+      baseURL: 'https://api.exampleaccesscontrol.com/manage',
+      method: 'GET',
+      headers: {
+        'User-Agent': 'Envoy Integration for Example Access Control',
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${Buffer.from(`${accessControlAppKey}:${accessControlAppSecret}`).toString('base64')}`,
```
Comment (shaunpersad) [inline lib/AccessControlClient.js]: can you use the axios `auth` option instead?

File: lib/AccessControlClient.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      },
+    });
+  }
+
+  async authenticate() {
+    return await this.orgList();
+  }
+
+  async authenticateDeprecated(envoyUserId) {
```
Comment (shaunpersad) [inline lib/AccessControlClient.js]: Do we need to keep this if it's deprecated?

File: lib/handlers/events/webhookInvite.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  const emailFormatValid = helper.Strings.validateEmail(email);
+  if (!emailFormatValid) {
+    await job.attach({ label: 'Email Error', value: 'The email provided for this visitor can not be accepted by Access Control.' });
+    return res.sendFailed('Access Control Error: Invalid Email');
+  }
+
+  try {
+    const AppKey = await installStorage.get('accessControlAppKey');
```
Comment (shaunpersad) [inline lib/handlers/events/webhookInvite.js:52]: This should be `appKey` and `appSecret`

File: lib/handlers/setupStep/setupStepAuthenticate.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        accessControlAppKey,
+        accessControlAppSecret,
+        accessControlOrgId,
+      },
+      installStorage,
+      userAPI,
+    }
+  } = req;
+
+  const storageAppKey    = await installStorage.get('accessControlAppKey');
```
Comment (shaunpersad) [inline lib/handlers/setupStep/setupStepAuthenticate.js]: Did eslint run on this file?

File: lib/handlers/events/webhookInvite.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      .filter(group => {
+        const { isGlobal } = group;
+        return `${isGlobal}` === 'false';
+      })
+      .map(group => {
+        return group.tag;
+      });
+
+    /* ******************************************
+     * begin building data blob for user creation
```
Comment (shaunpersad) [inline lib/handlers/events/webhookInvite.js]: This is not a blob

Comment (Stephen-Arsenault) [reply lib/handlers/events/webhookInvite.js]: omg...

**Review by shaunpersad — APPROVED** [review]

**Pull request merged**